src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -607,6 +609,51 @@ void CoreWorkerDirectTaskSubmitter::PushNormalTask(
       });
 }
 
+const ray::rpc::ClientCallback<ray::rpc::GetTaskFailureCauseReply>
+CoreWorkerDirectTaskSubmitter::CreateGetTaskFailureCauseCallback(
+    const Status &status,
+    const bool is_actor,
+    const TaskID &task_id,
+    const rpc::WorkerAddress &addr) {
+  return [this, status, is_actor, task_id, addr](
+             const Status &get_task_failure_cause_reply_status,
+             const rpc::GetTaskFailureCauseReply &get_task_failure_cause_reply) {
+    rpc::ErrorType task_error_type = rpc::ErrorType::WORKER_DIED;
```

Comment:
Call this method directly as the callback instead of having it return a callback?